pages/manageTemplates.tsx

```diff
@@ -0,0 +1,95 @@
+    timestamp: new Date('December 17, 1995 03:24:00'),
+    createdBy: 'mike'
+  }
+  const template2: MessageTemplate = {
+    messageID: 2,
+    title: 'title 2',
+    message: 'message 2',
+    timestamp: new Date('December 17, 1995 03:24:00'),
+    createdBy: 'karyna'
+  }
+
+  const template3: MessageTemplate = {
+    messageID: 3,
+    title: 'title 3',
+    message: 'message 3',
+    timestamp: new Date('October 9, 2003 03:24:00'),
+    createdBy: 'dean'
+  }
+
+  const messageArray: MessageTemplate[] = [template1, template2, template3]
+  return (
+        <>
+            <Layout>
+                <ThemeProvider theme={theme}>
+                    <StyledPageContainer>
+                        <Grid container spacing={2}>
+                            <Grid item xs={10}>
+                                <Typography variant="h4">Manage Templates</Typography>
+                            </Grid>
+                            <Grid item xs={2}>
+                                <Link href="/emailSender" underline="hover">
+                                    {'←'} Email Sender
+                                </Link>
+                            </Grid>
+                        </Grid>
+                        <Divider />
+                        <br />
+                        <SectionContainer>
+                            <TemplateDropdown templates={messageArray} />
+                        </SectionContainer>
+                        <SectionContainer>
+                            <Stack direction="row" spacing={1}>
+                                <Typography variant="body1">Template Name</Typography>
+                                <Typography color="red">*</Typography>
+                            </Stack>
+                            <br></br>
+                            <StyledTextField
+                                id="outlined-basic"
+                                variant="outlined"
+                                placeholder="Template Name"
+                            // onChange={handleEmailSubject}
+                            />
+                        </SectionContainer>
+                        <SectionContainer>
+                            <Stack direction="row" spacing={1}>
+                                <Typography variant="body1">Template Message</Typography>
+                                <Typography color="red">*</Typography>
+                            </Stack>
+                            <br></br>
+                            <StyledTextArea
+                                aria-label="message-text-area"
+                                placeholder="Paste in message"
+                                // onChange={(e) => setMessage(e.target.value)}
+                                minRows={5}
+                            />
+                        </SectionContainer>
+                        <Button variant="contained" component="span">
+                            Create New Message
```

Comment:
Could you make sure that both template name and template message have values when the button is pressed? A couple of ways to do this:
- disable the button until there is text in both fields, or
- see https://mui.com/material-ui/react-text-field/#validation - when a text field is invalid, use the `error` prop

Reply:
for checking if there is text in the fields, do you recommend using states, refs, or maybe something else?

Reply:
I would use state - use `onChange` to set the state as the user types, for each text field. You can check the text field where email body goes in the sender as an example

Reply:
omg ok i tried using onChange but it weirdly appears to be returning the previous event

https://user-images.githubusercontent.com/37753574/184299954-40fb050c-bf44-44c5-a277-cedc295e35b7.mov

Reply:
This is because state updates are asynchronous - so this log `console.log('CHANGE MADE ' + templateName)` is logging the previous state value. Generally you shouldn't access and update state in the same function. You can move the above log somewhere else (right above the return statement for example) and it should work fine

Reply:
i see!! i used 'useEffect' and it seemed to work! (i think it acts as a callback function?? although i am not sure how)

Reply:
looks good!